Tests/JSONAPITests/Relationships/RelationshipTests.swift

```diff
@@ -235,6 +235,37 @@ extension RelationshipTests {
 		test_DecodeEncodeEquality(type: ToManyWithMetaAndLinks.self,
 								  data: to_many_relationship_with_meta_and_links)
 	}
+
+  func test_ToManyRelationshipWithMetaNoDataOmittable() {
+    TestEntityType1.canHaveNoDataInRelationships = true
+    
+    let relationship = decoded(type: ToManyWithMeta.self,
+                               data: to_many_relationship_with_meta_no_data)
+
+    XCTAssertEqual(relationship.ids, [])
+    XCTAssertEqual(relationship.meta.a, "hello")
+
+    TestEntityType1.canHaveNoDataInRelationships = false
+  }
+
+  func test_ToManyRelationshipWithMetaNoDataNotOmittable() {
+    TestEntityType1.canHaveNoDataInRelationships = false
+
+    do {
+        _ = try decodedThrows(type: ToManyWithMeta.self,
+                              data: to_many_relationship_with_meta_no_data)
+        XCTFail("Expected decoding to fail.")
+    } catch let error as DecodingError {
+      switch error {
+      case .keyNotFound(ResourceLinkageCodingKeys.data, _):
+        break
+      default:
+        XCTFail("Expected error to be DecodingError.keyNotFound(.data), but got \(error)")
+      }
+    } catch {
+      XCTFail("Expected to have DecodingError.keyNotFound(.data), but got \(error)")
+    }
```

Comment:
I would prefer to keep the coding keys private, not do an `@testable import`, and test the description of the error instead of the exact coding key it outputs. This is just a preference of mine, I don't think there's a right or wrong way to go about testing this.

The test could use the `XCTAssertThrowsError` to clean up a bit of the `XCTFail` work done above and it would look something like:
```swift
    XCTAssertThrowsError(
      try decodedThrows(type: ToManyWithMeta.self,
                        data: to_many_relationship_with_meta_no_data)
    ) { error in 
      XCTAssertEqual(error.localizedDescription, "The data couldn’t be read because it is missing.")
    }
```